### ISSUE TITLE
Not reporting a gauge value of zero

### ISSUE THREAD

**Issue by isobit** [dialtone/metrics-exporter-statsd-full#5]
There seems to be no exported metric when explicitly reporting a gauge value of `0`, for example with `metrics::gauge!("foo", 0 as f64)`.

I noticed there were a lot of "if 0, continue" conditions added in https://github.com/dialtone/metrics-exporter-dogstatsd/commit/85bd93b4045581d92f7927201ac163ae3a573573, maybe using zero to indicate idle is breaking explicit zero values?

**Comment by dialtone** [comment]
yeah, not exactly sure how to manage that differently though. Easiest obviously is to simply let it be a configuration parameter whether 0 is reported or not, although probably you want a low idle timeout when doing that so you don't report the 0 each time... 

Do you have any opinion?

**Comment by ijc** [comment]
This is affecting us too. I think the problem is that 0 is a completely valid value for a gauge or a counter (e.g. "number of active sessions" can easily be zero and is definitely distinct from1) and zero samples in a period is legit for a histrogram.

Shouldn't idleness be dealt with by setting `idle_timeout` to eventually discard those metrics? Why do they need to be dealt with any further with filtering after that?
